Candidate draft (#1014)
nl-design-system/documentatie: docs/handboek/candidate-stappenplan.mdx

  - candidate
---

# Stappenplan: Candidate

> Bij **Candidate** is het component opgenomen in de `@nl-design-system-unstable` organisatie. Het kernteam verwacht dat het component uiteindelijk een Hall of Fame status krijgt. De API's zijn nog niet gegarandeerd en kunnen nog wijzigen. Ook bestaat er een mogelijkheid dat het component vervangen wordt door een vergelijkbare component. In deze fase bekijken we of het component mogelijk nog extra goed getest en stabiel gemaakt moet worden. Omdat het component door het kernteam onderhouden wordt voldoet het, naast de meeste Community DoD eisen, al wel aan veel meer criteria voor stabiel hergebruik.

Reviewer: Dit is dezelfde tekst als genoemd bij de Dod? Misschien is het goed om iets meer aan te sluiten op de intro teksten zoals geschreven zijn bij Help Wanted en [Community](https://github.com/nl-design-system/documentatie/blob/903d70eaa7f10e05337f83724f5e49b9e03b3745/docs/handboek/community-stappenplan.mdx#stappenplan-community)? Maar dan in de zin van...

'Het kernteam volgt dit stappenplan om een component van alle checkpoints 🚩 te voorzien die nodig zijn voor de 'Candidate' van het Estafettemodel.

Heb je een vraag? Twijfel niet en stel je vraag via het [#nl-design-system Slack kanaal](https://praatmee.codefor.nl/) van CodeForNL of [neem contact op met het kernteam](https://github.com/nl-design-system/documentatie/blob/903d70eaa7f10e05337f83724f5e49b9e03b3745/project/kernteam).

> ## 1. Voltooien van de eigenschappen

Reviewer: In de stappenplannen van Help Wanted en Community hebben we deze onderverdeling met headings uit de D.o.d. niet doorgevoerd. Ik vind het geen gek idee, dus als we dit willen doen dan trek ik dat ook graag recht in die stappenplannen.

```suggestion
## 1. Voltooien van de eigenschappen
```

### Versimpelen of opsplitsten

Doel: Component in lijn brengen met het doel en use case.

Wanneer een component diverse doelen en use cases dient, kan het kernteam ervoor kiezen om dit component op te splitsen of te vereenvoudigen. Het [doel](/handboek/help-wanted-stappenplan#doel-bepalen) is de basis waarop we componenten indelen.

**🚩 Checkpoint**: Versimpeld of gesplitst

### API volgt conventies

> Doel: API volgt conventies en is geprefixed met 'nl'.

Reviewer: Omdat wij dit Candidate stappenplan als kernteam doorlopen snap ik dat er wellicht minder toelichting nodig is. En toch vraag ik me af of het ook voor een communitylid fijn zou zijn om iets meer context te krijgen per checkpoint. Zoals bijvoorbeeld ook is gedaan bij 'Versimpelen of opsplitsen'.

**🚩 Checkpoint**: API gemeenschappelijk

### Component kan omgaan met verschillende soorten content

Doel: Het component is flexibel en kan verschillende soorten content aan, zoals korte tekst, veel woorden, lange woorden, of zelfs ontbrekende inhoud.

**🚩 Checkpoint**: Content flexibel

### Goedkeuring van de semantiek

Doel: De semantiek (roles, names, states in HTML/ARIA) is goedgekeurd door het kernteam.

**🚩 Checkpoint**: Semantiek goedgekeurd

### Gespiegelde werkelijkheid tussen Figma en code

Doel: De Figma-bibliotheek van het NL Design System moet overeenkomen met de implementatie van het component in code, met betrekking tot varianten, states, design tokens en naamgeving.

Een component kan gebruikmaken van enums (selectielijst in Figma), booleans (toggle in Figma) en strings (invoerveld in Figma). Het is mogelijk om de naam van de eigenschappen (properties) en het type eigenschap tussen Figma en de code-implementatie gelijk te trekken.

**🚩 Checkpoint**: Figma in lijn

### Linting en automatische code-formatting

Doel: De codebase maakt gebruik van linting en automatische codeformatting met Prettier volgens de configuraties van het NL Design System.

> De juister configuraties zijn gewaardborgd in de *example* [repository](https://github.com/nl-design-system/example).

Reviewer: Schrijffoutje en het woord 'Example' in de link opgenomen zodat het ook voor screenreaders duidelijker is waar deze link naartoe linkt.

```suggestion
De juiste configuraties zijn gewaardborgd in de [Example repository](https://github.com/nl-design-system/example).
```

**🚩 Checkpoint**: Linting en formatting

## 2. Tokens opleveren

### Gebruikt in verschillende huisstijlen

Doel: Het component is ingezet door twee of meer organisaties met verschillende huisstijlen en toont aan dat het met meerdere huisstijlen overweg kan. 

> Controleer of er genoeg eigenschappen zijn die via tokens aangepast kunnen worden om aan de behoeften van de huisstijlen van de betrokken organisaties te voldoen.

Reviewer: Hier begint de beschrijving met een werkwoord. Fijn! Dat zet aan tot actie. Ik ben als lezer wel benieuwd hoe ik dit concreet zou moeten aanpakken.

**🚩 Checkpoint**: Common tokens

## 3. Opleveren van de documentatie

### Aliassen van alternatieve namen

Doel: Alternatieve namen voor het component zijn gedocumenteerd als aliassen.

Dit maakt het vinden van componenten of alternatieven gemakkelijker. Aliassen en alternatieve namen ontstaan meestal tijdens de 'Help Wanted' fase. Voorbeelden hiervan zijn: _Spinner_ en _Loader_ of _Select_ en _Dropdown_.

**🚩 Checkpoint**: Documentatie aliassen

### Anatomie documenteren

Doel: De anatomie van het component is vastgelegd.

De anatomie verwijst naar de naam van het component en de onderdelen ervan, zoals de _button_ en het _label_ dat deel uitmaakt van de button.

**🚩 Checkpoint**: Documentatie anatomie

### Accessibility en inclusie documentatie

Doel: Accessibility en inclusie documentatie is beschikbaar voor de overwegingen die van toepassing zijn.

**🚩 Checkpoint**: Documentatie A11Y

## 4. Testen van het component

> ### Demonstreren en testen van features.

Reviewer: Zijn 'features' en 'configuraties' hetzelfde? Kunnen we iets minder engels en jargon ('Coverage features') gebruiken om deze stap helder neer te zetten voor iemand die minder ingevoerd is?

Doel: De Storybook-stories en Figma-voorbeelden demonstreren alle features die tevens zijn opgenomen als unit tests.

In Figma tonen we de mogelijke configuraties op een pagina. Bij Storybook kun je via voorselecties of de knobs de configuraties zien.

**🚩 Checkpoint**: Coverage features

### Getest op geschiktheid voor internationalisatie.

> Doel: De implementatie is getest op geschiktheid voor internationalisering om te verzekeren dat het component correct functioneert in verschillende taalomgevingen.

Reviewer: Hoe doe je dat? Dit is ook interessant voor lezers die hun Community component hier al op willen voorbereiden.

**🚩 Checkpoint**: I18N getest

### Op toegankelijkheid getest

> Doel: De implementatie en documentatie is op toegankelijkheid getest.

Reviewer: Hoe doe je dat? Dit is ook interessant voor lezers die hun Community component hier al op willen voorbereiden.

**🚩 Checkpoint**: A11Y getest

### Documentatie is getest

Doel: De documentatie is getest onder de doelgroepen en goed ontvangen.

> De documentatie is eenduidig, helder en volledig. Wanneer we iemand uit de doelgroep vragen of ze op basis van deze documentatie het component wel of niet zouden gebruiken, kunnen ze daarop een antwoord geven.

Reviewer: Kunnen we spreken over communityleden?

```suggestion
De documentatie is eenduidig, helder en volledig. Wanneer we communityleden vragen of ze op basis van deze documentatie het component wel of niet zouden gebruiken, kunnen ze daarop een antwoord geven.
```

**🚩 Checkpoint**: Documentatie getest

> 

Reviewer: Dit is een extra entertje. Daar ging de linting volgens mij van over zijn nek.

Doel: Verwijzing is beschikbaar naar extern gebruikersonderzoek die tot de keuze van dit component leiden.

**🚩 Checkpoint**: Documentatie onderzoek

### Gepubliceerd op NPM

Doel: Het component is gepubliceerd op NPM in de scope van `@nl-design-system-unstable`.

In de volgende fase, de 'Hall of Fame', komt het component beschikbaar in de NPM scope van @nl-design-system. In dat geval ontvang je een melding dat dit component _'deprecated'_ is binnen `*-unstable`.

**🚩 Checkpoint**: NPM (unstable)

### Changelog wordt gepubliceerd

Doel: Bij nieuwe versies van dit component wordt de changelog op Github gepubliceerd, zodat gebruikers goed op de hoogte blijven van wijzigingen.

Zoals de [button van Utrecht](https://github.com/nl-design-system/utrecht/blob/main/components/button/CHANGELOG.md)

**🚩 Checkpoint**: Changelog

### Status bijgewerkt naar Candidate.

Doel: De status is bijgewerkt naar naar Candidate.

Het component heeft alle stappen van Candidate doorlopen. Het is een stabiele versie die inhoudelijk niet meer veranderd. We zijn nu klaar voor de volgende fase: de [Hall of Fame](https://nldesignsystem.nl/handboek/hall-of-fame-stappenplan).

> 

Reviewer: Ik mis de promotie hier nog. Kan volgens mij gebaseerd worden op de promotie van [Community](https://github.com/nl-design-system/documentatie/blob/903d70eaa7f10e05337f83724f5e49b9e03b3745/docs/handboek/community-stappenplan.mdx#promotie).

Ohh het eerste echte nl. component. Ik ga echt een tattoo zetten van die classname.
